server/aesdsocket.c

```diff
@@ -0,0 +1,232 @@
+
+	if(remove(SOCK_DATA_FILE) != 0) {
+		fprintf(stderr, "Error deleting %s\n", SOCK_DATA_FILE);
+		exit(1);
+	}
+
+	exit(0);
+}
+
+// Check if string contains char c
+// return 1 if it does, otherwise 0
+int contains_chr(char *str, char c, int len) {
+	for(int i=0; i<len; i++) {
+		if(str[i] == c) {
+			return 1;
+		}
+	}
+
+	return 0;
+}
+
+int main(int argc, char *argv[])
+{
+	// Step 1: Setup
+
+	// Part A: Set up addrinfo
+
+	struct addrinfo hints;
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_flags = AI_PASSIVE;
+	hints.ai_protocol = 0;
+	hints.ai_addr = NULL;
+	int status;
+
+	if((status = getaddrinfo(NULL, SOCK_SERV_PORT, &hints, &g_servinfo)) != 0) {
+		fprintf(stderr, "getaddrinfo err: %s\n", gai_strerror(status));
+		return 1;
+	}
+
+	// Part B: Set up Syslog
+	openlog(NULL, LOG_PID | LOG_CONS | LOG_PERROR, LOG_USER);
+
+	// Part C: Register signal handlers
+	signal(SIGINT, handle_kill);
+	signal(SIGTERM, handle_kill);
+
+	// Step 2: Create socket fd
+
+	g_sockfd = socket(PF_INET, SOCK_STREAM, 0);
+
+	if(g_sockfd == SOCK_ERR) {
+		freeaddrinfo(g_servinfo);
+		fprintf(stderr, "socket err\n");
+		closelog();
+		return 1;
+	}
+
+	// note: line taken from https://stackoverflow.com/questions/24194961/how-do-i-use-setsockoptso-reuseaddr
+	if (setsockopt(g_sockfd, SOL_SOCKET, SO_REUSEADDR, &(int){1}, sizeof(int)) < 0) {
+		freeaddrinfo(g_servinfo);
+		fprintf(stderr, "setsockopt err\n");
+		closelog();
+		return 1;
+	}
+
+	// Step 3: Bind the socket
+
+	int bind_res = bind(g_sockfd, g_servinfo->ai_addr, sizeof(struct sockaddr));
+
+	if(bind_res == SOCK_ERR) {
+		freeaddrinfo(g_servinfo);
+		fprintf(stderr, "bind err\n");
+		closelog();
+		return 1;
+	}
+
+	// Step 3a: Run as daemon if requested
+	if(argc > 1) {
+		if(strcmp(argv[1], "-d") == 0) {
+			daemon(0, 0);
+		}
+	}
+
+	// Step 4: Listen for socket connections
+
+	int listen_res = listen(g_sockfd, MAX_CONNECTIONS);
+
+	if(listen_res == SOCK_ERR) {
+		freeaddrinfo(g_servinfo);
+		fprintf(stderr, "listen err\n");
+		closelog();
+		return 1;
+	}
+
+	// Step 5: Accept and handle connections
+
+	struct sockaddr *client_sockaddr = NULL;
+	socklen_t sockaddr_len = 0;
+	g_data_file = NULL;
+	int recv_size = 0;
+	int recv_size_tot = 0;
+	int contains_newlin = 0;
+	g_dyn_recv_buf = NULL;
+
+	while(1) {
+		// Accept connection
+		g_accepted_sock_fd = accept(g_sockfd, client_sockaddr, &sockaddr_len);
+
+		if(g_accepted_sock_fd == SOCK_ERR) {
+			fprintf(stderr, "socket accept err\n");
+			continue;
+		}
+
+		// Log IP of accepted connection
+		syslog(LOG_DEBUG, "Accepted connection from %s\n", inet_ntoa(((struct sockaddr_in*)&client_sockaddr)->sin_addr));
+
+		// receive data
+		recv_size_tot = 0;
+		char recv_buf[RECV_BUF_LEN] = {0};
+		while(1) {
+			recv_size = (int) recv(g_accepted_sock_fd, recv_buf, RECV_BUF_LEN, 0);
+			recv_size_tot += recv_size;
+			g_dyn_recv_buf = realloc(g_dyn_recv_buf, recv_size_tot);
```

Comment:
Good!  I think you could probably just use `g_dyn_recv_buf` here reather than adding another `recv_buf` and avoid the memset/memcpy though.